### ISSUE TITLE
Structured facts display could be improved

### ISSUE THREAD

**Issue by gertvdijk** [voxpupuli/puppetboard#83]
[Facter 2.0 introduced structured facts](http://docs.puppetlabs.com/facter/2.0/release_notes.html) (compared to only key-value pairs in 1.x). In the 2.0.x series of Facter, no structured facts were provided by default. However, as stated in the [Facter 2.1.0 announcement](https://groups.google.com/d/msg/puppet-users/sbwKLwLa4r0/_Xab9lmLfroJ), now `partitions` is the first standard structured fact (hash).

Please add support for this in Puppetboard. By support I mean one should think of a way it can be displayed 'nicely', rather than for example:

```
partitions  {"vda1"=>{"uuid"=>"af740304-b2c6-440f-877a-5fb15cf86c39", "size"=>"12578816", "filesystem"=>"ext4"}}
```

**Comment by corey-hammerton** [comment]
Starting with PuppetDB 3.0 Structured facts are available in JSON format and new endpoints fact-paths, fact-contents and factsets are available. 

As @daenney said it is a lot of work to implement reliable browsing in PuppetBoard with both string and JSON type fact values. This will be on the backlog but if anybody really needs this support please submit a PR with a link to this issue.

**Comment by daenney** [comment]
So Puppetboard has a `jsonprint` template filter that you can use, see `templates/metric.html` and `templates/metric.html`.

It doesn't provide for a nice way to browse and click through it but will at least print it as more humanly readable JSON instead of a long string.

**Comment by corey-hammerton** [comment]
We are open to PRs if anyone wants to attempt to implement an API for this. Hopefully it can make it into the 0.2.x series.

**Comment by corey-hammerton** [comment]
As of PuppetDB 4.2.0 a new Inventory endpoint is being provided as well as "dotted query" syntax is made available, also the dot notation is experimental.

It may be possible to make use of these new features in the 0.3.0 series with PuppetDB 5.0.0.

**Comment by gdubicki** [comment]
Fixed in v3.3.0 released today. :)
